### PR TITLE
Feature/private label leaderboard

### LINKED ISSUE #1959: Show Private Label for Leaderboard entries with maypublish = false
In Leaderboard,
If someone tries to open a profile for a `user without a slug` or user who has `maypublish=false`

Show the following label:
Screen Shot 2020-02-15 at 8 58 17 PM

Note that, label `should not show` initially when the leaderboard tab is loaded.

https://zpl.io/V4dLEEJ

### PR DESCRIPTION
**Pull request by haideralishah**
Fixes #1959

Changes in this pull request:
- Add `PRIVATE` label in all leaderboard

@sagararyal

### CODE REVIEW

File: app/components/LeaderboardRefresh/Companies/CompaniesLeaderBoard.js [hunk 1]
```diff
@@ -81,9 +81,16 @@ const CompaniesLeaderBoard = ({ navigation }) => {
...
-                  <Text numberOfLines={2} style={styles.countryNameText}>
-                    {item.caption}
-                  </Text>
+                  <View style={styles.countryNameCont}>
+                    <Text numberOfLines={2} style={styles.countryNameText}>
+                      {item.caption}
+                    </Text>
+                    {!isPrivate ? null : (
+                      <View>
+                        <Text style={styles.privateText}>PRIVATE</Text>
```
Comment (norbertschuler) [inline app/components/LeaderboardRefresh/Companies/CompaniesLeaderBoard.js]: I would suggest to make a translation resource of the word "PRIVATE" as in German is would be named "PRIVAT".

File: app/components/LeaderboardRefresh/Countries/CountriesLeaderBoard.js [hunk 2]
```diff
@@ -77,9 +77,16 @@ const CountriesLeaderBoard = ({ navigation }) => {
...
-                  <Text numberOfLines={2} style={styles.countryNameText}>
-                    {item.caption}
-                  </Text>
+                  <View style={styles.countryNameCont}>
+                    <Text numberOfLines={2} style={styles.countryNameText}>
+                      {item.caption}
+                    </Text>
+                    {!isPrivate ? null : (
+                      <View>
+                        <Text style={styles.privateText}>PRIVATE</Text>
```
Comment (norbertschuler) [inline app/components/LeaderboardRefresh/Countries/CountriesLeaderBoard.js]: Same as above.

File: app/components/LeaderboardRefresh/Countries/CountryDetails.js [hunk 3]
```diff
@@ -87,9 +87,16 @@ const CountryDetails = ({ navigation }) => {
...
-                  <Text numberOfLines={2} style={styles.countryNameText}>
-                    {item.caption}
-                  </Text>
+                  <View style={styles.countryNameCont}>
+                    <Text numberOfLines={2} style={styles.countryNameText}>
+                      {item.caption}
+                    </Text>
+                    {!isPrivate ? null : (
+                      <View>
+                        <Text style={styles.privateText}>PRIVATE</Text>
```
Comment (norbertschuler) [inline app/components/LeaderboardRefresh/Countries/CountryDetails.js]: Same as above.

File: app/components/LeaderboardRefresh/Individuals/IndividualsLeaderBoard.js [hunk 4]
```diff
@@ -82,9 +82,16 @@ const IndividualsLeaderBoard = ({ navigation }) => {
...
-                  <Text numberOfLines={2} style={styles.countryNameText}>
-                    {item.caption}
-                  </Text>
+                  <View style={styles.countryNameCont}>
+                    <Text numberOfLines={2} style={styles.countryNameText}>
+                      {item.caption}
+                    </Text>
+                    {!isPrivate ? null : (
+                      <View>
+                        <Text style={styles.privateText}>PRIVATE</Text>
```
Comment (norbertschuler) [inline app/components/LeaderboardRefresh/Individuals/IndividualsLeaderBoard.js]: Same as above.

File: app/components/LeaderboardRefresh/Schools/SchoolsLeaderBoard.js [hunk 5]
```diff
@@ -81,9 +81,16 @@ const SchoolsLeaderBoard = ({ navigation }) => {
...
-                  <Text numberOfLines={2} style={styles.countryNameText}>
-                    {item.caption}
-                  </Text>
+                  <View style={styles.countryNameCont}>
+                    <Text numberOfLines={2} style={styles.countryNameText}>
+                      {item.caption}
+                    </Text>
+                    {!isPrivate ? null : (
+                      <View>
+                        <Text style={styles.privateText}>PRIVATE</Text>
```
Comment (norbertschuler) [inline app/components/LeaderboardRefresh/Schools/SchoolsLeaderBoard.js]: Same as above.

File: app/components/LeaderboardRefresh/TPOs/tpoLeaderBoard.js [hunk 6]
```diff
@@ -81,9 +81,16 @@ const tpoLeaderBoard = ({ navigation }) => {
...
-                  <Text numberOfLines={2} style={styles.countryNameText}>
-                    {item.caption}
-                  </Text>
+                  <View style={styles.countryNameCont}>
+                    <Text numberOfLines={2} style={styles.countryNameText}>
+                      {item.caption}
+                    </Text>
+                    {!isPrivate ? null : (
+                      <View>
+                        <Text style={styles.privateText}>PRIVATE</Text>
```
Comment (norbertschuler) [inline app/components/LeaderboardRefresh/TPOs/tpoLeaderBoard.js]: Same as above (sorry for repeating me, but we have this much duplicated source code in the Leaderboard)

Comment (haideralishah) [reply app/components/LeaderboardRefresh/TPOs/tpoLeaderBoard.js]: fixed

**Pull request merged**